### PR TITLE
[CDAP-6174] Feature/ui new log viewer refinement

### PR DESCRIPTION
**Pull request by PGuimarais**
- Enables full screen
- Adds polling logic
  - If number of logs returned does not fill table, and current program is running, poll for new logs
  - Stop polling when start time is changed
- Style fixes

Bamboo build: http://builds.cask.co/browse/CDAP-DRC4018-19

### CODE REVIEW

File: cdap-ui/app/directives/log-viewer/log-viewer.html [hunk 1]
```diff
@@ -67,7 +66,7 @@
     Loading Logs...
   </div>
   <!-- LOGS TABLE -->
-  <div ng-if="LogViewer.data.length > 0 && !LogViewer.loading" class="logs-table">
+  <div ng-if="LogViewer.data.length > 0 && !LogViewer.loading" class="logs-table" ng-class="{'table-full-screen': LogViewer.fullScreen}">
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.html]: the ng-class here is unnecessary. You already set the full-screen class on the container, you should be able to target the child element from that

File: cdap-ui/app/directives/log-viewer/log-viewer.html [hunk 2]
```diff
@@ -94,24 +102,24 @@
                 ng-click="LogViewer.collapseColumns()"></span>
           <span>Message</span>
           <div class="expand-all-btn text-center"
-               ng-click="LogViewer.toggleLogExpansion()">Expand All</div>
+               ng-click="LogViewer.toggleLogExpansion()"><span ng-if="!LogViewer.toggleExpandAll">Expand All</span><span ng-if="LogViewer.toggleExpandAll">Collapse All</span></div>
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.html]: separate these on newlines for readability

File: cdap-ui/app/directives/log-viewer/log-viewer.html [hunk 3]
```diff
@@ -136,11 +145,11 @@
               <div class="icon-column"
                    ng-if="(entry.log.stackTrace.length > 0 && entry.isStackTraceExpanded)"
                    ng-class="Logviewer.message-class">
-                <span class="fa"
+                <span class="fa message-arrow"
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.html]: is message-arrow being used?

File: cdap-ui/app/directives/log-viewer/log-viewer.js [hunk 4]
```diff
@@ -56,7 +61,30 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi) {
...
+  });
+
+  this.logEvents = ['ERROR', 'WARN', 'INFO', 'DEBUG', 'TRACE'];
+
+  let included = {
+    'ERROR' : false,
+    'WARN' : false,
+    'INFO' : false,
+    'DEBUG' : false,
+    'TRACE' : false
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.js:76]: shouldn't the default be everything true?

File: cdap-ui/app/directives/log-viewer/log-viewer.js [hunk 5]
```diff
@@ -168,18 +211,27 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi) {
...
+        this.viewLimit = 100;
+        this.cacheDecrement = 100;
+        this.cacheSize = 0;
+
+        //If there are no logs yet to be retrieved let viewer know
+        if(res.length < this.viewLimit){
+          this.loadingMoreLogs = false;
+        } else {
+          this.loadingMoreLogs = true;
+        }
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.js]: Use tertiary

File: cdap-ui/app/directives/log-viewer/log-viewer.less [hunk 6]
```diff
@@ -27,15 +27,87 @@
 @stack-trace-font-color: #cccccc;
 @message-expanded-color: #dedede;
 @searchbar-placeholder-color: #787878;
-@searchbar-input-color: blue;
+@searchbar-input-color: #9cbcfc;
 @timeline-buffer-color: #a6a6a6;
 @error-color: #f65454;
 @warning-color: #f6b354;
 
+body.state-flows-detail {
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.less]: is the styling of the tooltip going to be different across different states?

File: cdap-ui/app/directives/log-viewer/log-viewer.less [hunk 7]
```diff
@@ -27,15 +27,87 @@
...
+  .source-tooltip.tooltip {
+    .tooltip-inner {
+      background-color: black;
+      border: 1px solid black;
+      color: white;
+      padding: 7px;
+      word-wrap: break-word;
+      text-align: left;
+      .border-radius(3px);
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.less]: just use `border-radius: 3px`

File: cdap-ui/app/directives/log-viewer/log-viewer.less [hunk 8]
```diff
@@ -27,15 +27,87 @@
...
+      bottom: 0;
+      left: 0;
+      right: 0;
+      overflow: none;
+      background-color: white;
+      z-index: -50;
+    }
+  }
+
+  .highlightedText {
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.less]: dash delimited

**Comment by ajainarayanan** [comment]
There is a JS error on minified code 
screen shot 2016-07-18 at 10 52 44 am

File: cdap-ui/app/directives/log-viewer/log-viewer.js (hunk 4 above)
Comment (PGuimarais) [inline cdap-ui/app/directives/log-viewer/log-viewer.js:76]: This is referring to checkbox filtering. In this instance, all event types have yet to be checked, thus they default to false.

File: cdap-ui/app/directives/log-viewer/log-viewer.less (hunk 6 above)
Comment (PGuimarais) [inline cdap-ui/app/directives/log-viewer/log-viewer.less]: No, it should remain the same.

File: cdap-ui/app/directives/log-viewer/log-viewer.js (hunk 4 above)
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.js:76]: Isn't that wrong? The default is you are showing all log levels. That means everything should be selected

File: cdap-ui/app/directives/log-viewer/log-viewer.less (hunk 6 above)
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.less]: then you don't need the `.state-flows-detail`

File: cdap-ui/app/directives/log-viewer/log-viewer.js (hunk 4 above)
Comment (PGuimarais) [inline cdap-ui/app/directives/log-viewer/log-viewer.js:76]: The default behavior is that all logs are shown, while none of the checkboxes are selected. When the user chooses to filter based on specific event types, they can elect to 'include' whichever event type they'd like in the filtering. By default, there are no event types included in the filtering - which is why their values are defaulted to false. 

The primary purpose is to narrow down the entries to those the user is looking for - given this behavior, to view a specific event type requires a single click, whereas the inverse would require 4 clicks to unselect the others to view a single event type. 

If a user wants to view 4 event types, it requires 4 clicks. If they want to view a single event, it requires a single click. This seems much more intuitive than the inverse.

**Comment by elfenheart** [comment]
screen shot 2016-07-18 at 4 54 51 pm

Moving the slider I'm getting this error. It's querying the wrong API. Going to another program still retains the old program as well

**Comment by ajainarayanan** [comment]
Checked for errors in my local after the update. Worked fine. Still lot of corner case we need to address and those will be coming in subsequent PRs. Hence merging it for now 👍 LGTM

**Pull request merged**